### PR TITLE
Glossary

### PR DESCRIPTION
**Pull request by shiroyasha**
One challenge I was working on today was to make the terminology for our new Tests tab crystal clear.

During this process, I realized that having this terminology specified in the docs 
will help both us internally, and our customers externally.

@markoa I would like to hear your thoughts about this proposal.

### CODE REVIEW

File: docs/reference/glossary.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+The result of a single test case with one of the following outcomes:
+
+- Passed - The test passed the assertions in the test case
+- Failed - The test failed the assertions in the test case
+- Error - The test case failed to execute. Assertion was not tested.
+- Timeout - The test case failed to execute in preset interval. Assertion was not tested.
+- Skipped - The test case was skipped, and has no result.
+
+#### Test results
```
Comment (markoa) [inline docs/reference/glossary.md:62]: At this point the terminology begins to smell. :) _Test result_ and _Test results_ are too similar as terms for what they refer to.

Perhaps ~~Test result~~ → Test case result && ~~Test results~~ → Test suite results?

Comment (markoa) [reply docs/reference/glossary.md:62]: Depending on what you want to do here, there may be changes to the final section of the doc.

Comment (shiroyasha) [reply docs/reference/glossary.md]: I feel that we would do ourselves a disservice if we didn't cover the term "test results" on this page.
We commonly use this term in various contexts, it must be covered.

I decided to merge these two sections into one unified `test results` term and cover both the singular and plural form.

**Comment by shiroyasha** [comment]
I've covered the feedback points + I did a round of Grammarly on the text.

@markoa looking forward to your review.

**Review by markoa — APPROVED** [review]
LGTM!

Now we just have to backfill all older terms. 😅

**Pull request merged**